### PR TITLE
Update charm/v8 to remove series restrictions on kubernetes bundles.

### PR DESCRIPTION
**Pull request by hpidcock**
Allow k8s bundles to specify series for sidecar charms. Podspec charms will still need to specify `series: kubernetes`.

## QA steps

```
juju download kubeflow
unzip kubeflow_*
# edit bundle.yaml add series: bionic or focal or something to a sidecar based app
juju deploy ./bundle.yaml --trust
```

## Documentation changes

N/A

## Bug reference

N/A

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -181,6 +181,7 @@ require (
 	github.com/juju/gojsonpointer v0.0.0-20150204194629-afe8b77aa08f // indirect
 	github.com/juju/gojsonreference v0.0.0-20150204194633-f0d24ac5ee33 // indirect
 	github.com/juju/lru v0.0.0-20190314140547-92a0afabdc41 // indirect
+	github.com/juju/mgo/v3 v3.0.3 // indirect
```
Comment (barrettj12) [inline go.mod:184]: How did this sneak in here? Do we want it?

File: go.sum [hunk 2]
```diff
@@ -498,15 +498,16 @@ github.com/juju/ansiterm v1.0.0 h1:gmMvnZRq7JZJx6jkfSq9/+2LMrVEwGwt7UR6G+lmDEg=
...
-github.com/juju/charm/v8 v8.0.5/go.mod h1:tZ0JfWOdv11qu4Gm5lPD0KHBeuVUH2vbrKFyYS6JUAw=
+github.com/juju/charm/v8 v8.0.6 h1:fTm1Kn3JHNVbdsetPas1icMy+MUoCHrpfEW5eqWKlTY=
+github.com/juju/charm/v8 v8.0.6/go.mod h1:tZ0JfWOdv11qu4Gm5lPD0KHBeuVUH2vbrKFyYS6JUAw=
 github.com/juju/charmrepo/v6 v6.0.3 h1:gSdJk0Hc+g3nktVzrs86/YtKIbD9ClOXZ7WpvxYNeCE=
 github.com/juju/charmrepo/v6 v6.0.3/go.mod h1:3bgmJGGixQ205agoj0oFYvCj+VTzDxSQhR20QmUxJJI=
 github.com/juju/clock v0.0.0-20190205081909-9c5c9712527c/go.mod h1:nD0vlnrUjcjJhqN5WuCWZyzfd5AHZAC9/ajvbSx69xA=
 github.com/juju/clock v0.0.0-20220202072423-1b0f830854c4/go.mod h1:zDZCPSgCJQINeZtQwHx2/cFk4seaBC8Yiqe8V82xiP0=
 github.com/juju/clock v0.0.0-20220203021603-d9deb868a28a/go.mod h1:GZ/FY8Cqw3KHG6DwRVPUKbSPTAwyrU28xFi5cqZnLsc=
-github.com/juju/clock v1.0.0 h1:9U1jPr/FT0DfZjl8qh9dSACAXUodjPOz+uZBdeVv/N8=
 github.com/juju/clock v1.0.0/go.mod h1:GZ/FY8Cqw3KHG6DwRVPUKbSPTAwyrU28xFi5cqZnLsc=
```
Comment (barrettj12) [inline go.sum:508]: try `go mod tidy` again, probably this line should have been removed

File: go.sum [hunk 3]
```diff
@@ -562,6 +563,8 @@ github.com/juju/mgo/v2 v2.0.0-20210302023703-70d5d206e208/go.mod h1:0OChplkvPTZ1
 github.com/juju/mgo/v2 v2.0.0-20220111072304-f200228f1090/go.mod h1:N614SE0a4e+ih2rg96Vi2PeC3cTpUOWgCTv3Cgk974c=
 github.com/juju/mgo/v2 v2.0.2 h1:ufYtW2OFNjniTuxOngecP3Mk5sSclo8Zl1mnmyGWUWA=
 github.com/juju/mgo/v2 v2.0.2/go.mod h1:Z2QbXIrR9JuJcSyankQOw31tINNA5p3qevW73oDoHsM=
+github.com/juju/mgo/v3 v3.0.3 h1:z0H0OVB+lWujviaw2Zsxwn553kGoS7a03eyh3Erf2jY=
+github.com/juju/mgo/v3 v3.0.3/go.mod h1:fAvhDCRbUlEbRIae6UQT8RvPUoLwKnJsBgO6OzHKNxw=
```
Comment (barrettj12) [inline go.sum:567]: yeah I don't think we want mgo/v2 and mgo/v3 right?

File: go.sum [hunk 4]
```diff
@@ -637,8 +640,9 @@ github.com/juju/version v0.0.0-20191219164919-81c1be00b9a6/go.mod h1:kE8gK5X0CIm
 github.com/juju/version v0.0.0-20210303051006-2015802527a8 h1:BTo6HzRR0zPBcXbs1Sy08aQNfvdm3ey8O+mBTiO3g00=
 github.com/juju/version v0.0.0-20210303051006-2015802527a8/go.mod h1:YUOK6VK9+V4hpmsBYx8ymvOzEXRzoIGjRDKpIezTi60=
 github.com/juju/version/v2 v2.0.0-20211007103408-2e8da085dc23/go.mod h1:Ljlbryh9sYaUSGXucslAEDf0A2XUSGvDbHJgW8ps6nc=
-github.com/juju/version/v2 v2.0.0 h1:0OjCOlgA2UBbB8fSCgd8fmGsntrSbAWXEJil47bQptY=
 github.com/juju/version/v2 v2.0.0/go.mod h1:ZeFjNy+UFEWJDDPdzW7Cm9NeU6dsViGaFYhXzycLQrw=
```
Comment (barrettj12) [inline go.sum:643]: Again should be gonezo

File: go.mod (hunk 1 above)
Comment (hpidcock) [reply go.mod:184]: its not actually used

**Review by wallyworld — APPROVED** [review]

**Pull request merged**